Frameworks/Foundation/NSIndexPath.mm

```diff
@@ -38,22 +49,36 @@ + (NSIndexPath*)indexPathWithIndexes:(const NSUInteger*)indexes length:(NSUInteg
  @Status Interoperable
 */
 + (NSIndexPath*)indexPathForRow:(NSInteger)row inSection:(NSInteger)section {
-    unsigned indexes[] = { section, row };
+    if (row < 0 || section < 0) {
```

Comment:
>row < 0 || section < 0) { [](start = 8, length = 25)

is this valid? given the fact that they are passing in a NSInteger vs NSUInteger.
is this the expected behavior on ref platform?